public/app/features/dashboard/dashboard_model.ts

```diff
@@ -370,6 +387,40 @@ export class DashboardModel {
         }
       }
     }
+
+    // Update gridPos for panels below
+    if (selectedOptions.length > 1) {
+      let panelBelowIndex = panelIndex + selectedOptions.length;
+      for (let i = panelBelowIndex; i < this.panels.length; i++) {
+        this.panels[i].gridPos.y += yPos;
+      }
+    }
+  }
+
+  repeatCollapsedPanel(panel: PanelModel, panelIndex: number, repeatPanel: PanelModel) {
```

Comment:
why do we need to repeat collapsed panels?

Reply:
To recalc hidden panels and correct view number on collanped row. To show on First row "(1 hidden panels)" instead "(2 hidden panels)" in this situation
![06](https://user-images.githubusercontent.com/936530/35098980-0b7a7808-fc78-11e7-8d38-8e1b53ec88cf.jpg)